.github/workflows/publish-nuget.yaml

```diff
@@ -19,8 +20,23 @@ jobs:
       - name: Get version number
         id: version
         uses: PaulHatch/semantic-version@v5.2.1
-        with:
-          version_format: "${major}.${minor}.${patch}-beta${increment}"
+
+      - name: Generate semver version number
+        id: generate-version
+        shell: powershell
+        run: |
+          if("${{ steps.version.outputs.is_tagged }}" -eq "false")
+          {
+            $assembly_version = "${{ steps.version.outputs.major }}.${{ steps.version.outputs.minor }}.${{ steps.version.outputs.patch }}.${{ steps.version.outputs.increment }}"
+            $semver_version = "${{ steps.version.outputs.major }}.${{ steps.version.outputs.minor }}.${{ steps.version.outputs.patch }}-beta.${{ steps.version.outputs.increment }}"
+          }
+          else
+          {
+            $assembly_version = "${{ steps.version.outputs.major }}.${{ steps.version.outputs.minor }}.${{ steps.version.outputs.patch }}.${{ steps.version.outputs.increment }}"
+            $semver_version = "${{ steps.version.outputs.major }}.${{ steps.version.outputs.minor }}.${{ steps.version.outputs.patch }}"
+          }
+          echo ::set-output name=assembly-version::${assembly_version}
```

Comment:
This way of setting outputs is deprecated. The recommended way is
```powershell
echo assembly-version=${assembly_version} >> $env:GITHUB_OUTPUT
echo semantic-version=${semver_version} >> $env:GITHUB_OUTPUT
```

Reply:
Thanks for this --> done